saleor/graphql/checkout/mutations/utils.py

```diff
@@ -274,24 +276,49 @@ def get_checkout(
     return checkout
 
 
-def group_quantity_and_custom_prices_by_variants(
-    lines: List[Dict[str, Any]]
-) -> List[CheckoutLineData]:
+def group_lines(lines: List[Dict[str, Any]]) -> List[CheckoutLineData]:
     variant_checkout_line_data_map: Dict[str, CheckoutLineData] = defaultdict(
         CheckoutLineData
     )
+    force_new_line_checkout_lines_data_map: List[CheckoutLineData] = []
 
     for line in lines:
+        variant_db_id = None
+        line_db_id = None
         variant_id = cast(str, line.get("variant_id"))
-        line_data = variant_checkout_line_data_map[variant_id]
+        line_id = cast(str, line.get("line_id"))
+        force_new_line = line.get("force_new_line")
+
+        if line_id:
+            _, line_db_id = graphene.Node.from_global_id(line_id)
+
+        if variant_id:
+            _, variant_db_id = graphene.Node.from_global_id(variant_id)
+            variant_db_id = int(variant_db_id)
+
+        if force_new_line:
+            line_data = CheckoutLineData(variant_id=variant_db_id)
+            force_new_line_checkout_lines_data_map.append(line_data)
+        else:
+            if line_id:
+                line_data = variant_checkout_line_data_map[line_id]
```

Comment:
I don't get why we need the `variant_checkout_line_data_map`? We are never assigning anything to it, and only getting the value from default dict which always will be `CheckoutLineData`, cannot we just define the `line_data` at the beginning of the for loop?

Reply:
Also, the name of the variable suggests that the key in this map is `variant` but we use `line_id` and `variant_id` as keys.